src/App.tsx

```diff
@@ -1,61 +1,111 @@
+import React, { FormEventHandler, useState } from 'react';
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+
+import { Todo, User } from './types/index';
+import { TodoList } from './components/TodoList/TodoList';
+
+export function getUser(userId: number): User | null {
+  const foundUser = usersFromServer.find(user => user.id === userId);
+
+  return foundUser || null;
+}
+
+export const todos: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUser(todo.userId),
+}));
+
+export const App: React.FC = () => {
+  const [title, setTitle] = useState('');
+  const [userId, setUserId] = useState<number>(0);
+  const [formSubmitted, setFormSubmitted] = useState(false);
+  const [updatedTodos, setUpdatedTodos] = useState(todos);
+
+  const handleSubmit: FormEventHandler = (event) => {
+    event.preventDefault();
+    if (title && userId) {
+      const newTodo = {
+        id: Math.max(...updatedTodos.map(todo => todo.id)) + 1,
+        title,
+        completed: false,
+        userId: Number(userId),
+        user: getUser(Number(userId)),
+      };
+
+      setUpdatedTodos([...updatedTodos, newTodo]);
+      setFormSubmitted(false);
+      setTitle('');
+      setUserId(0);
+    } else {
+      setFormSubmitted(true);
+    }
+  };
 
-export const App = () => {
   return (
     <div className="App">
       <h1>Add todo form</h1>
 
-      <form action="/api/todos" method="POST">
+      <form action="/api/todos" onSubmit={handleSubmit} method="POST">
+
         <div className="field">
-          <input type="text" data-cy="titleInput" />
-          <span className="error">Please enter a title</span>
+          <label htmlFor="title">
+            Title:
+            {' '}
```

Comment:
Remove this line and use template strings.

[More info.](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals)